### ISSUE TITLE
[Regression] PPU loading never finishes for some titles (#13534)

### ISSUE THREAD

**Issue by SlyCooperReloadCoded** [RPCS3/rpcs3#13815]
### Quick summary

Starting with PR 13534, the loading screens for PPU caches on some games never complete when they should.

### Details

Starting with PR 13534, the loading screens for PPU caches on some games never complete when they should, sitting on "Linking PPU Modules" indefinitely, but without anything crashing.  Manually closing will cause a deadlock.  Watch Dogs is a perfect example and what I used for my logs.

![Capture](https://user-images.githubusercontent.com/58268914/236635033-5d582af1-44a2-40d1-9837-86cb91beefa2.PNG)

### Build with regression

v0.0.27-14846-fb88e1c1

### Attach two log files

Log ffrom PR 13534 (v0.0.27-14846-fb88e1c1):
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/11412764/RPCS3.log)

Log from PR 13613 (v0.0.27-14845-7081b89e) ⁪:
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/11412767/RPCS3.log)

### System configuration

Intel Core i9-10900K
NVIDIA RTX 2080 SUPER
64GB DDR4 Memory
Windows 10 Pro x64 Build 19043

**Comment by SlyCooperReloadCoded** [comment]
Apparently if you wait a veeeeery long time, it makes it ingame... only to be met with issue #13723.

**Comment by AniLeo** [comment]
Seems related to #13877

**Comment by Darkhost1999** [comment]
Yes, this is the same issue reported before the other 1 was posted. Yet the other 1 is more informative that this issue is more widespread than 1 game.
Watchdogs may be one of the most afflicted.

**Comment by AniLeo** [comment]
Worth retesting with #14090

**Comment by SlyCooperReloadCoded** [comment]
Nope, still an issue as of latest master.

**Comment by SlyCooperReloadCoded** [comment]
This still happens for me on latest master, and on every game in my library, Watch Dogs taking the longest.  Just hangs there at "Linking PPU modules" for up to 10 minutes with nothing printing to the log, just like before.

**Comment by Darkhost1999** [comment]
I believe LLVM 17 will be here in October. Not just a release candidate but hopefully an actual release. Maybe it'll get resolved with us bumping LLVM when the time comes.

**Comment by AniLeo** [comment]
Could also be a good idea to update to 16.0.5 in the meantime, since we're using 16.0.1

**Comment by AniLeo** [comment]
Please retest

**Comment by FlexBy420** [comment]
Watch Dogs being booting just fine for some time now.
[RPCS3.log.gz](https://github.com/user-attachments/files/19490291/RPCS3.log.gz)
